--- SQLite.g4	2020-02-26 10:38:32.079377494 +0100
+++ SQLite_new.g4	2020-02-26 10:40:46.620469963 +0100
@@ -31,18 +31,11 @@
 grammar SQLite;
 
 parse
- : ( sql_stmt_list | error )* EOF
- ;
-
-error
- : UNEXPECTED_CHAR 
-   { 
-     throw new RuntimeException("UNEXPECTED_CHAR=" + $UNEXPECTED_CHAR.text); 
-   }
+ : ( sql_stmt_list )+ (SINGLE_LINE_COMMENT | MULTILINE_COMMENT_EOF)?
  ;
 
 sql_stmt_list
- : ';'* sql_stmt ( ';'+ sql_stmt )* ';'*
+ : sql_stmt ( ';'+ sql_stmt )*
  ;
 
 sql_stmt
@@ -862,8 +855,12 @@
  : '--' ~[\r\n]* -> channel(HIDDEN)
  ;
 
+MULTILINE_COMMENT_EOF
+ : '/*' .*? -> channel(HIDDEN)
+ ;
+
 MULTILINE_COMMENT
- : '/*' .*? ( '*/' | EOF ) -> channel(HIDDEN)
+ : '/*' .*? '*/' -> channel(HIDDEN)
  ;
 
 SPACES
